### PR TITLE
Feature/844 cross axisa axisb

### PR DESCRIPTION
**Pull request by ClaudiaComito**
## Description

Introducing changes to accomodate numpy's `axisa`, `axisb`, `axisc` into @mtar 's `ht.cross` implementation.

Still TODO: documentation update, expand on tests, memory/performance checks.
 
Issue/s resolved: #

## Changes proposed:
-
-
-
-

## Type of change

- New feature (non-breaking change which adds functionality)
## Due Diligence
- [ ] All split configurations tested
- [ ] Multiple dtypes tested in relevant functions
- [ ] Documentation updated (if needed)
- [ ] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/linalg/basics.py [hunk 1]
```diff
@@ -74,45 +82,88 @@ def cross(x1: DNDarray, x2: DNDarray, axis: int = -1) -> DNDarray:
...
-    if not isinstance(axis, int):
-        try:
-            axis = int(axis)
-        except Exception:
-            raise TypeError("'axis' must be an integer.")
+    x1_2d, x2_2d = False, False
+    x1_shape, x2_shape = list(x1.shape), list(x2.shape)
 
-    axis = stride_tricks.sanitize_axis(x1.shape, axis)
+    if not axis == -1 or torch.unique(torch.tensor([axisa, axisb, axisc, axis])).numel() == 1:
```
Comment (mtar) [inline heat/core/linalg/basics.py:95]: If the default value of `axis` is set to `None`, the check will be shortened.

Comment (ClaudiaComito) [reply heat/core/linalg/basics.py:95]: `axis = None` is not compatible with the [Python array API](https://data-apis.org/array-api/latest/extensions/linear_algebra_functions.html?highlight=cross#linalg-cross-x1-x2-axis-1) though.

If we leave `axis=-1` as default, we're compatible with the array API and `cross` will still work as a numpy drop-in replacement.

Comment (ClaudiaComito) [reply heat/core/linalg/basics.py:95]: > If the default value of `axis` is set to `None`, the check will be shortened. Also, some redundant code can be avoided when the axes parameter are set beforehand to the value of it.
> 
> ```python
> if axis is not None:
>     axis = stride_tricks.sanitize_axis(x1.shape, axis)
>     axisa, axisb, axisc = (axis,) * 3
> ```

You're right about the redundant code though, I replaced it with your suggestion.
